pkg/ibmvpc/vpc.go

```diff
@@ -866,40 +867,94 @@ func actionNameStr(action *string) (string, error) {
 	return denyAction, nil
 }
 
-func (tgw *TransitGateway) StringPrefixDetails(src, dst vpcmodel.Node, verbose bool) (string, error) {
-	prefix := tgw.prefixOfSrcDst(src, dst)
-	transitEnablesConn := vpcmodel.HasNode(tgw.sourceNodes, src) && vpcmodel.HasNode(tgw.destNodes, dst)
-	if verbose {
-		tgwRouterFilterDetails, err := tgw.tgwPrefixStr(*prefix)
-		if err != nil {
-			return "", err
-		}
-		action := "blocks"
-		if transitEnablesConn {
-			action = "allows"
-		}
-		return fmt.Sprintf("transit gateway %s %s connection with the following prefix\n\t%s\n\n",
-			tgw.Name(), action, tgwRouterFilterDetails), nil
-	}
-	noVerboseStr := fmt.Sprintf("cross-vpc-connection: transit-connection %s of transit-gateway %s ", *prefix.tc.Name, tgw.Name())
-	if transitEnablesConn {
-		return noVerboseStr + "allows connection" + newline, nil
-	}
-	return noVerboseStr + "denies connection" + newline, nil
-}
-
-func (tgw *TransitGateway) prefixOfSrcDst(src, dst vpcmodel.Node) *tgwPrefixFilter {
+// RulesInConnectivity returns the prefix filters relevant for <src, dst>. Since src/dst could be a cidr,
+// there could be more than one relevant prefix filter (in a single transit connection)
+// todo: currently "more than one" is not possible since src/dst must be within a given subnet.
+// todo This will be relevant only once we allow src/dst to span over vpcs
+// todo https://github.com/np-guard/vpc-network-config-analyzer/issues/576
+func (tgw *TransitGateway) RulesInConnectivity(src, dst vpcmodel.Node) []vpcmodel.RulesInTable {
```

Comment:
I'll separate it to two separate functions, one for the verbose case and the other for the non-verbose. Hope this will help. 

Since the printing relies on actual IBM's transit gateway and transit connections, I'm not sure how unittests can be written here with a reasonable amount of effort; non unit tests already exists. Perhaps we can take it online.

Reply:
I was referring to the function of the computation  `RulesInConnectivity`, not the printing function

Reply:
There is an example for unit tests with transit connections at `tgw_analysis_test.go`.
We can discuss what is required for basic unit tests here.

Reply:
Turns out most of the code here was redundant - since for transit gateway based routing each atomic src/dst is an endpoint with a single IP address. 
I added a test for tgw in which a cidr with a number of endpoints in it is given (and verified the test)